### PR TITLE
Clean up Kafka supervisor topic

### PR DESCRIPTION
**Pull request by ektravel**
### Description

This PR:
- places `intermediateHandoffPeriod` after `maxTotalRows`
- documents `skipBytesInMemoryOverheadCheck`
- removes future tense
- removes most of passive voice

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -1187,7 +1187,7 @@ There are additional configs for autoscaling (if it is enabled):
 |`druid.supervisor.idleConfig.enabled`|If `true`, supervisor can become idle if there is no data on input stream/topic for some time.|false|
 |`druid.supervisor.idleConfig.inactiveAfterMillis`|Supervisor is marked as idle if all existing data has been read from input topic and no new data has been published for `inactiveAfterMillis` milliseconds.|`600_000`|
 
-The `druid.supervisor.idleConfig.*` specified in the runtime properties of the overlord defines the default behavior for the entire cluster. See [Idle Configuration in Kafka Supervisor IOConfig](../development/extensions-core/kafka-supervisor-reference.md#kafkasupervisorioconfig) to override it for an individual supervisor.
+The `druid.supervisor.idleConfig.*` specified in the runtime properties of the overlord defines the default behavior for the entire cluster. See [Idle Configuration in Kafka Supervisor IOConfig](../development/extensions-core/kafka-supervisor-reference.md#supervisor-io-configuration) to override it for an individual supervisor.
```
Comment (techdocsmith) [inline docs/configuration/index.md:1190]:
```suggestion
The `druid.supervisor.idleConfig.*` specified in the Overlord runtime properties defines the default behavior for the entire cluster. See [Idle Configuration in Kafka Supervisor IOConfig](../development/extensions-core/kafka-supervisor-reference.md#supervisor-io-configuration) to override it for an individual supervisor.
```

File: docs/development/extensions-core/kafka-supervisor-reference.md [hunk 2]
```diff
@@ -186,64 +198,45 @@ Supported `inputFormat`s include:
...
+|`maxBytesInMemory`|Long|The number of bytes to aggregate in heap memory before persisting. This is based on a rough estimate of memory usage and not actual usage. Normally, this is computed internally. The maximum heap memory usage for indexing is `maxBytesInMemory * (2 + maxPendingPersists)`.|No|One-sixth of max JVM memory|
+|`skipBytesInMemoryOverheadCheck`|Boolean|The calculation of `maxBytesInMemory` takes into account overhead objects created during ingestion and each intermediate persist. To exclude the bytes of these overhead objects from the `maxBytesInMemory` check, set `skipBytesInMemoryOverheadCheck` to `true`.|No|`false`|
+|`maxRowsPerSegment`|Integer|The number of rows to aggregate into a segment; this number is post-aggregation rows. Handoff occurs when `maxRowsPerSegment` or `maxTotalRows` is reached or every `intermediateHandoffPeriod`, whichever happens first.|No|5000000|
```
Comment (techdocsmith) [inline docs/development/extensions-core/kafka-supervisor-reference.md:211]:
```suggestion
|`maxRowsPerSegment`|Integer|The number of rows to aggregate into a segment; this number is post-aggregation rows. Handoff occurs when `maxRowsPerSegment` or `maxTotalRows` is reached or every `intermediateHandoffPeriod`, whichever happens first.|No|5000000|
```
not for this PR, but I feel like "aggregate" into a segment is not right. Aggregation implies the combining of multiple rows into a single row. Here, individual rows make it into the segment. I think the number of rows to store in a segment would be closer.

**Review by techdocsmith — APPROVED** [review]
approved after nits

File: docs/development/extensions-core/kafka-supervisor-reference.md (hunk 2 above)
Comment (ektravel) [inline docs/development/extensions-core/kafka-supervisor-reference.md:211]:
```suggestion
|`maxRowsPerSegment`|Integer|The number of rows to store in a segment. This number is post-aggregation rows. Handoff occurs when `maxRowsPerSegment` or `maxTotalRows` is reached or every `intermediateHandoffPeriod`, whichever happens first.|No|5000000|
```

File: docs/development/extensions-core/kafka-supervisor-reference.md [hunk 3]
```diff
@@ -137,12 +148,12 @@ The following example demonstrates supervisor spec with `lagBased` autoScaler an
...
 
-In some cases, you may need to fetch consumer properties at runtime. For example, when `bootstrap.servers` is not known upfront, or is not static. To enable SSL connections, you must provide passwords for `keystore`, `truststore` and `key` secretly. You can provide configurations at runtime with a dynamic config provider implementation like the environment variable config provider that comes with Druid. For more information, see [DynamicConfigProvider](../../operations/dynamic-config-provider.md).
+In some cases, you may need to fetch consumer properties at runtime. For example, when `bootstrap.servers` is Not kNown upfront, or is Not static. To enable SSL connections, you must provide passwords for `keystore`, `truststore` and `key` secretly. You can provide configurations at runtime with a dynamic config provider implementation like the environment variable config provider that comes with Druid. For more information, see [Dynamic config provider](../../operations/dynamic-config-provider.md).
```
Comment (ektravel) [inline docs/development/extensions-core/kafka-supervisor-reference.md:156]:
```suggestion
In some cases, you may need to fetch consumer properties at runtime. For example, when `bootstrap.servers` is not known upfront, or is not static. To enable SSL connections, you must provide passwords for `keystore`, `truststore` and `key` secretly. You can provide configurations at runtime with a dynamic config provider implementation like the environment variable config provider that comes with Druid. For more information, see [Dynamic config provider](../../operations/dynamic-config-provider.md).
```

**Pull request merged**